src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java

```diff
@@ -1687,8 +1688,16 @@ private UrlEncodedFormEntity createUrlEncodedFormEntity(final String contentEnco
                 // Must decode the value now, so that when the
                 // httpclient encodes it, we end up with the same value
                 // as the user had entered.
-                parameterName = URLDecoder.decode(parameterName, urlContentEncoding);
-                parameterValue = URLDecoder.decode(parameterValue, urlContentEncoding);
+                try {
+                    URLCodec urlCodec = new URLCodec(contentEncoding);
+                    //parameterName = URLDecoder.decode(parameterName, urlContentEncoding);
```

Comment:
Don't leave old code as comments lying around. We have version control to look up old code structures :)